File: 0055-jump-game/0055-jump-game.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        maxindex=0
        
        for i in range(len(nums)):
            if(i>maxindex):
                return False
            maxindex=max(maxindex,i+nums[i])
        
        return True
        
        

File: 0055-jump-game/0055-jump-game.cpp
class Solution {
public:
    bool canJump(vector<int>& nums) {
        int maxindex=0;
       for(int i=0;i<nums.size();i++){
           if(i>maxindex)
               return 0;
           maxindex=max(maxindex,i+nums[i]);
       }
        return 1;
        
    }
};